purchases/src/test/java/com/revenuecat/purchases/backend_integration_tests/ProductionVirtualCurrenciesIntegrationTest.kt

```diff
@@ -0,0 +1,82 @@
+package com.revenuecat.purchases.backend_integration_tests
+
+
+import com.revenuecat.purchases.virtualcurrencies.VirtualCurrencies
+import org.assertj.core.api.Assertions.fail
+import org.junit.Test
+
+internal class ProductionVirtualCurrenciesIntegrationTest: BaseBackendIntegrationTest() {
+    override fun apiKey() = Constants.apiKey
+
+    @Test
+    fun `can fetch virtual currencies with a balance of 0`() {
+        ensureBlockFinishes { latch ->
+            backend.getVirtualCurrencies(
+                appUserID = "integrationTestUserWithAllBalancesEqualTo0",
+                appInBackground = false,
+                onSuccess = { virtualCurrencies ->
+                    validateVirtualCurrenciesObject(
+                        virtualCurrencies = virtualCurrencies,
+                        testVCBalance = 0,
+                        testVC2Balance = 0
+                    )
+                    latch.countDown()
+                },
+                onError = { error ->
+                    fail("Expected success but got error: $error")
+                },
+            )
+        }
+    }
+
+    @Test
+    fun `can fetch virtual currencies with a balance of greater than 0`() {
+        ensureBlockFinishes { latch ->
+            backend.getVirtualCurrencies(
+                appUserID = "integrationTestUserWithAllBalancesNonZero",
+                appInBackground = false,
+                onSuccess = { virtualCurrencies ->
+                    validateVirtualCurrenciesObject(
+                        virtualCurrencies = virtualCurrencies,
+                        testVCBalance = 100,
+                        testVC2Balance = 777,
+                    )
+                    latch.countDown()
+                },
+                onError = { error ->
+                    fail("Expected success but got error: $error")
+                },
+            )
+        }
+    }
+
+    private fun validateVirtualCurrenciesObject(
+        virtualCurrencies: VirtualCurrencies,
+        testVCBalance: Int,
+        testVC2Balance: Int,
+        testVC3Balance: Int = 0,
+    ) {
+        assert(virtualCurrencies.all.count() == 3)
+
+        val testCurrency = virtualCurrencies["TEST"]
```

Comment:
Nitpick but I wonder if we should create an entire expected VirtualCurrency object and compare it against each of these... That way we would notice if new fields are added...

Reply:
Love this! Implemented this here: 1f255d51e7e932432083d0598efe5874bde2679d